src/rail/estimation/algos/deepdisc.py

```diff
@@ -456,36 +508,38 @@ def _process_chunk(self, start_idx, metadata):
             The list of metadata dictionaries for this block of images
         """
 
-        print("Matching objects")
-        true_zs, pdfs = get_matched_z_pdfs(
-            metadata,
-            DC2ImageReader(),
-            lambda dataset_dict: dataset_dict["filename"],
-            self.predictor,
-        )
-        pdfs = np.array(pdfs)
-        num_pdfs = len(pdfs)
+        # createa a queue to receive output from the parallelized function
+        q = mp.Queue()
 
-        # don't write out if no pdfs are returned from the model
-        if num_pdfs == 0:
-            print("No PDFs returned from the model, skipping this chunk")
-            return
+        # call detectron2's `launch` function to parallelize the inference
+        launch(
+            _do_inference,
+            num_gpus_per_machine=2, #! correct this to be a StageParam
+            num_machines=1, #! correct this to be a StageParam
+            machine_rank=0, #? Need to figure out how to get this, maybe from the comm? or ceci???
```

Comment:
This might just be a copy/paste artifact, but do things go wrong if we use the _process_chunk param `rank` for this? It is familiar to me that ceci can and does keep track of machine ranks, but of course integrating ceci with pytorch is absolutely uncharted territory so I could see this being wonky

Reply:
Not quite copy/paste, but intentional hardcoding just to match the configuration that I've got for experimenting. I think that you're right, and this should use the param `rank`.